notebooks/COS/Extract/Extract.ipynb

```diff
@@ -0,0 +1,1580 @@
+   "source": [
+    "with open(outputdir+'output_calcos_fuv_1.txt', 'w') as f: # This file now contains the output of the last cell\n",
+    "    f.write(cap.stdout)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "attractive-patient",
+   "metadata": {},
+   "source": [
+    "**Now let's compare the FUV spectra extracted with the original and new XTRACTABs:**\n",
+    "\n",
+    "*Note* again, that we can ignore the UnitsWarning."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "early-reliance",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set up figure\n",
+    "fig = plt.figure(figsize = (20, 7))\n",
+    "gs = fig.add_gridspec(nrows = 5, ncols = 2) # Using gridspec to let us control panel sizes and locations\n",
+    "\n",
+    "for i,j in zip([0,1],[1,0]): # use i,j to plot FUVA on the right, not the left, by inverting subplot index\n",
+    "    ax0 = fig.add_subplot(gs[0:4,j])\n",
+    "    ax1 = fig.add_subplot(gs[4:5,j])\n",
+    "    new_wvln, new_flux = Table.read('./output/calcos_fuv_run1/le4b01040_x1dsum.fits')[i]['WAVELENGTH', 'FLUX']\n",
+    "    old_wvln, old_flux, seg = Table.read(old_x1dsum)[i]['WAVELENGTH', 'FLUX', 'SEGMENT']\n",
+    "    \n",
+    "    # Interpolate the new wvln onto the old wvln's sampling:\n",
+    "    new_flux_interp = interp1d(x = new_wvln, y = new_flux, fill_value=\"extrapolate\")(old_wvln)\n",
+    "\n",
+    "    # give max difference:\n",
+    "    print(f\"Stripe {seg} differs by up to: \\\n",
+    "    {max(new_flux - old_flux)/np.mean(abs(old_flux)):.3f}%\")\n",
+    "\n",
+    "    # Plotting - upper panel\n",
+    "    ax0.plot(new_wvln,new_flux,\n",
+    "             alpha = 1, linewidth = .5, c = 'C1',\n",
+    "             label = '$New$ extracted spectrum')\n",
+    "    ax0.plot(old_wvln, old_flux, linewidth = 1, c = 'C0',\n",
+    "             linestyle = 'dotted', alpha = 0.75,\n",
+    "             label = '$Original$ extracted spectrum')\n",
+    "    # Plotting - lower panel\n",
+    "    ax1.plot(new_wvln,old_flux - new_flux_interp, linewidth = .5, label = 'Residual')\n",
+    "    # Some formatting:\n",
+    "    ax0.set_title(f\"Segment {seg}\", fontsize = 20)\n",
+    "    ax0.set_xticks([])\n",
+    "    ax0.legend(loc = 'lower center')\n",
+    "    ax1.legend(loc = 'lower center')\n",
+    "    if i == 0: # Add axis labels to the plot\n",
+    "        ax0.set_ylabel(\"Flux\\n[$erg\\ \\AA^{-1}\\ cm^{-2}\\ s^{-1}$]\", fontsize = 20)\n",
+    "    if i == 1:\n",
+    "        plt.xlabel(\"Wavelength\", fontsize = 20)\n",
+    "plt.suptitle(\"Fig 4.4\\nComparing the old and new extracted spectra for each segment in the FUV\", fontsize = 25)\n",
+    "plt.tight_layout()\n",
+    "plt.savefig(plotsdir+\"comp_fuv_extracted.png\", dpi = 200)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "interim-texas",
+   "metadata": {},
+   "source": [
+    "## Congratulations! You finished this notebook!\n",
```

Comment:
As with the last one, I suggest you give at least a bit of a conclusion detailing what was "learned".